shoov/modules/custom/shoov_ci_build/drush/shoov_ci_build.drush.inc

```diff
@@ -99,4 +106,115 @@ function drush_shoov_ci_build_release_stuck_items() {
     drush_log(dt('@delta / @count) Re-queued message ID @id after being stuck.', $params));
     $delta++;
   }
-}
\ No newline at end of file
+}
+
+/**
+ * Release items that are marked in progress for too long.
+ */
+function drush_shoov_ci_build_release_failed_items() {
+
+  $range = 50;
+
+  $query = new EntityFieldQuery();
+  $result = $query
+    ->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'ci_incident')
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->propertyCondition('created', time() - 3*24*60*60, '<')
+    ->fieldCondition('field_ci_build_system_disabled', 'value', FALSE)
+    ->addTag('fixed_build_empty')
+    ->propertyOrderBy('changed', 'ASC')
+    ->addTag('DANGEROUS_ACCESS_CHECK_OPT_OUT')
+    ->range(0, $range)
+    ->execute();
+
+  if (empty($result['node'])) {
+    return;
+  }
+
+  $ids = array_keys($result['node']);
+  $nodes = node_load_multiple($ids);
+
+  foreach ($nodes as $node) {
+    drush_print($node->nid);
+
+    // Incident wrapper.
+    $wrapper = entity_metadata_wrapper('node', $node);
+
+    // CI Build wrapper
+    $build_wrapper = $wrapper->field_ci_build;
+
+    if (!$build_wrapper->field_ci_build_enabled->value()) {
+      // Build is already disabled.
+      $wrapper->field_reminders_sent->set(0);
+      $wrapper->field_ci_build_system_disabled->set(1);
+      $wrapper->save();
+      continue;
+    }
+
+    $timestamp = $node->created;
+
+    if ($timestamp > time() - 6*24*60*60 && $wrapper->field_reminders_sent->value() < 1) {
```

Comment:
move each logic to own function. If one of those functions returns TRUE you know it means a mail should be sent.